application/classes/Ushahidi/Repository/Tag.php

```diff
@@ -174,12 +174,13 @@ public function deleteTag($id)
 	public function isRoleValid(Validation $validation, $fullData)
 	{
 		$valid = true;
-		$isChild = $fullData['parent_id'];
```

Comment:
@willdoran had to refactor this. 
1. I had a dumb bug checking $valid incorrectly, I suspect I  had tested and then changed something. Aaaanyway. FIxed now
2. I realized I needed to use entities otherwise we need to wrangle the fulldata role formats and we really shouldn't need to do that! 
3. Fixed tests to actually, you know.. test the correct thing.